extension/src/plots/model/index.ts

```diff
@@ -171,6 +182,58 @@ export class PlotsModel extends ModelWithPersistence {
     }
   }
 
+  public getCustomPlots() {
+    if (!this.customPlots) {
+      return
+    }
+    return {
+      plots: this.customPlots,
+      size: this.getPlotSize(Section.CUSTOM_PLOTS)
+    }
+  }
+
+  public recreateCustomPlots() {
+    const customPlots: CustomPlotData[] = collectCustomPlotsData(
+      this.getCustomPlotsOrder(),
+      this.experiments.getExperiments()
+    )
+    this.customPlots = customPlots
+  }
+
+  public getCustomPlotsOrder() {
+    return this.customPlotsOrder
+  }
+
+  public setCustomPlotsOrder(plotsOrder: CustomPlotsOrderValue[]) {
+    this.customPlotsOrder = plotsOrder
+    this.persist(PersistenceKey.PLOTS_CUSTOM_ORDER, this.customPlotsOrder)
+    this.recreateCustomPlots()
+  }
+
+  public removeCustomPlots(plotIds: string[]) {
+    const newCustomPlotsOrder = this.getCustomPlotsOrder().filter(
+      ({ metric, param }) => {
+        return !plotIds.includes(getCustomPlotId(metric, param))
+      }
+    )
+
+    this.setCustomPlotsOrder(newCustomPlotsOrder)
+  }
+
+  public addCustomPlot(metricAndParam: CustomPlotsOrderValue) {
+    const plotAlreadyExists = this.getCustomPlotsOrder().some(
+      ({ param, metric }) =>
+        param === metricAndParam.param && metric === metricAndParam.metric
+    )
+
+    if (plotAlreadyExists) {
+      return Toast.showError('Custom plot already exists.')
```

Comment:
[I] Ideally this class is responsible for data/data manipulation. `Plots` should use `Toast` and inform the user when there is a problem.

Reply:
> Maybe we could remove the choice from the list instead of letting them select something that already exists.

Definitely doable by looping over the custom plots and checking for what exists already!